internal/cloud/realm/user_ext_test.go

```diff
@@ -0,0 +1,195 @@
+					switch user.Identities[0].ProviderType {
+					case realm.AuthProviderTypeUserPassword:
+						emailUsers = append(emailUsers, user)
+					case realm.AuthProviderTypeAPIKey:
+						apiKeyUsers = append(apiKeyUsers, user)
+					}
+				}
+
+				assert.Equal(t, 3, len(emailUsers))
+				assert.Equal(t, []realm.User{email1, email2, email3}, emailUsers)
+
+				assert.Equal(t, 2, len(apiKeyUsers))
+				for _, user := range apiKeyUsers {
+					identity := user.Identities[0]
+					_, ok := apiKeyIDs[identity.UID]
+					assert.True(t, ok, "expected %s to match a previously created api key id", identity.UID)
+					apiKeyIDs[identity.UID] = user.ID
+				}
+			})
+
+			t.Run("And find a certain type of user", func(t *testing.T) {
+				users, err := client.FindUsers(groupID, app.ID, realm.UserFilter{Providers: []realm.AuthProviderType{realm.AuthProviderTypeUserPassword}})
+				assert.Nil(t, err)
+				assert.Equal(t, []realm.User{email1, email2, email3}, users)
+			})
+
+			t.Run("And find specific user ids", func(t *testing.T) {
+				users, err := client.FindUsers(groupID, app.ID, realm.UserFilter{IDs: []string{email2.ID, email3.ID}})
+				assert.Nil(t, err)
+				assert.Equal(t, []realm.User{email2, email3}, users)
+			})
+
+			t.Run("And disable users", func(t *testing.T) {
+				assert.Nil(t, client.DisableUser(groupID, app.ID, email1.ID))
+				assert.Nil(t, client.DisableUser(groupID, app.ID, email3.ID))
+			})
+
+			t.Run("And find all disabled users", func(t *testing.T) {
+				users, err := client.FindUsers(groupID, app.ID, realm.UserFilter{State: realm.UserStateDisabled})
+				assert.Nil(t, err)
+
+				email1.Disabled = true
+				email3.Disabled = true
+
+				assert.Equal(t, []realm.User{email1, email3}, users)
+			})
+
+			t.Run("And find specific user using all filter options", func(t *testing.T) {
+				filter := realm.UserFilter{
+					IDs:       []string{email2.ID, email3.ID, apiKeyIDs[apiKey1.ID]},
+					State:     realm.UserStateDisabled,
+					Providers: []realm.AuthProviderType{realm.AuthProviderTypeUserPassword},
+				}
+				users, err := client.FindUsers(groupID, app.ID, filter)
+				assert.Nil(t, err)
+				assert.Equal(t, []realm.User{email3}, users)
+			})
+
+			t.Run("And revoking a user session should succeed", func(t *testing.T) {
+				assert.Nil(t, client.RevokeUserSessions(groupID, app.ID, email1.ID))
+			})
+
+			t.Run("And delete users", func(t *testing.T) {
+				for _, userID := range []string{email1.ID, email2.ID, email3.ID, apiKeyIDs[apiKey1.ID], apiKeyIDs[apiKey2.ID]} {
+					assert.Nilf(t, client.DeleteUser(groupID, app.ID, userID), "failed to successfully delete user: %s", userID)
+				}
+			})
+		})
+
+		t.Run("And finding pending users should return an empty list", func(t *testing.T) {
+			users, err := client.FindUsers(groupID, app.ID, realm.UserFilter{Pending: true})
+			assert.Nil(t, err)
+			assert.Equal(t, []realm.User{}, users)
+		})
+	})
+}
+
+func TestAuthProviderTypeDisplay(t *testing.T) {
```

Comment:
I would move this and `TestNewAuthProviderTypes` to the `user_test.go` file instead